addons/mail/controllers/message_reaction.py

```diff
@@ -5,13 +5,23 @@
 from odoo import http
 from odoo.http import request
 from odoo.addons.mail.models.discuss.mail_guest import add_guest_to_context
+from odoo.addons.mail.tools.discuss import Store
 
 
 class MessageReactionController(http.Controller):
     @http.route("/mail/message/reaction", methods=["POST"], type="json", auth="public")
     @add_guest_to_context
-    def mail_message_add_reaction(self, message_id, content, action):
+    def mail_message_add_reaction(self, message_id, content, action, **kwargs):
         message = request.env["mail.message"].browse(int(message_id)).exists()
         if not message._validate_access_for_current_persona("write"):
             raise NotFound()
-        message.sudo()._message_reaction(content, action)
+        partner, guest = self._compute_reaction_author(message, **kwargs)
+        if not partner and not guest:
+            raise NotFound()
+        store = Store()
+        # sudo: mail.message - access mail.message.reaction through an accessible message is allowed
+        message.sudo()._message_reaction(content, action, partner, guest, store)
+        return store.get_result()
+
+    def _compute_reaction_author(self, message, **kwargs):
+        return request.env["res.partner"]._get_current_persona()
```

Comment:
I'm not fully sure yet to keep this API, and it is a bit confusing as part of this PR. I would keep it in the main PR for now.

This PR should just do the store change/param, and ok to pass `partner, guest`, but they should directly call `_get_current_persona`, not introduce this extra method.